dynawo/sources/Models/Modelica/Dynawo/Electrical/Machines/GeneratorPVAlphaSum.mo

```diff
@@ -0,0 +1,73 @@
+within Dynawo.Electrical.Machines;
+
+/*
+* Copyright (c) 2015-2020, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source time domain simulation tool for power systems.
+*/
+
+model GeneratorPVAlphaSum "Model for generator PV with AlphaSum."
+
+  extends BaseClasses.BaseGeneratorSimplified;
+  extends AdditionalIcons.Machine;
+
+  public
+
+    parameter Types.PerUnit KGover = 1 "Mechanical power sensitivity to frequency";
+    parameter Types.PerUnit Lambda = 0 "Parameter Lambda of the voltage regulation";
+    parameter Types.ActivePowerPu PMinPu "Minimum active power in p.u (base SnRef)";
+    parameter Types.ActivePowerPu PMaxPu "Maximum active power in p.u (base SnRef)";
+    parameter Types.ReactivePowerPu QMinPu  "Minimum reactive power in p.u (base SnRef)";
+    parameter Types.ReactivePowerPu QMaxPu  "Maximum reactive power in p.u (base SnRef)";
+    parameter Types.ActivePowerPu PAuxHVPu = 0 "Active power consumption of the high voltage auxiliary used to recalculate the P and Q limits when the auxiliaries are not explicitly modeled (base SnRef)";
+    parameter Types.ActivePowerPu QAuxHVPu = 0 "Reactive power consumption of the high voltage auxiliary used to recalculate the P and Q limits when the auxiliaries are not explicitly modeled (base SnRef)";
+    parameter Types.ActivePowerPu PAuxLVPu = 0 "Active power consumption of the low voltage auxiliary used to recalculate the P and Q limits when the auxiliaries are not explicitly modeled  (base SnRef)";
+    parameter Types.ActivePowerPu QAuxLVPu = 0 "Reactive power consumption of the low voltage auxiliary used to recalculate the P and Q limits when the auxiliaries are not explicitly modeled  (base SnRef)";
+    final parameter Types.ActivePowerPu PMinAuxPu = PMinPu - PAuxHVPu - PAuxLVPu "Minimum active power in p.u when taking into account the auxiliaries (base SnRef)";
```

Comment:
Could we only have PMinPu PMaxPu QMinPu and QMaxPu in the dynamic model and do the limits calculation in the init model?

Reply:
I could, but in the dynamic model I would have PMinAuxPu and not PMinPu (same for Max and Q). Is that ok ?

Reply:
In the init model you can use PMin, PMax, PAuxHV, PAuxLV and calculate PMinPu and PMax Pu for the dynamic model, would it work that way?

Reply:
I would like to keep things in pu as much as possible (like it is done with auxHV_P0Pu in Dynawo voltage stability for example).  But if you think it is better, I am ok to go with your solution.